DEVDOCS: Improving the page "create-cache-type" (#6680)
magento/devdocs: src/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.md

The tag *scope* provides a mechanism for a cache type.

## Adding a custom cache type {#m2devgde-adding-custom-cache-type}

This topic discusses how to create a custom cache type.

### Cache type configuration {#m2devgde-cache-type-configuration}

Declare a new cache type in the `etc/cache.xml` file with the following attributes:

| Attribute | Required? | Description |
| --- | --- | --- |
| `name` | Yes | A unique cache type ID |

> | `translate` | No | Parameters that would be translated during to render "Cache Management" page |

Reviewer: We may add a reference to the paragraph above for clarity 

```suggestion
| `translate` | No | Parameters (described below) that will be translated on the "Cache Management" page |
```

Author: Hello @rogyar. Thank you for your remark. I fixed it. Please check.

> | `instance` | Yes | The cache type model |

Reviewer: ```suggestion
| `instance` | Yes | The cache type model class |
```

Author: Hello @rogyar. Thank you for your remark. I fixed it. Please check.

| --- | --- |
| `label` | The "Cache Type" field to be displayed on the `System` > `Tools` > `Cache Management` page. |
| `description` | The "Description" field to be displayed on the `System` > `Tools` > `Cache Management` page. |

For example:

```xml
<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Cache/etc/cache.xsd">
    <type name="%cache_type_id%" translate="label,description" instance="VendorName\ModuleName\Model\Cache\Type\CacheType">
        <label>Cache Type Label</label>
        <description>Cache Type Description</description>
    </type>
</config>
```

We’ve made a possibility that enables you to declare one or more cache types.

### Cache type model {#m2devgde-cache-type-model}

```php
<?php
namespace VendorName\ModuleName\Model\Cache\Type;

use Magento\Framework\App\Cache\Type\FrontendPool;
use Magento\Framework\Cache\Frontend\Decorator\TagScope;

/**
 * System / Cache Management / Cache type "Your Cache Type Label"
 */
class CacheType extends TagScope
{
    /**
     * Cache type code unique among all cache types
     */
    const TYPE_IDENTIFIER = '%cache_type_id%';

    /**

>      * Cache tag used to distinguish the cache type from all other cache

Reviewer: We may clarify this part since the explanation sounds pretty similar to the `TYPE_IDENTIFIER`. I would say that the cache tag is used for the cache scope (for cleaning operations as a practical example). See the [TagScope](https://github.com/magento/magento2/blob/2.3/lib/internal/Magento/Framework/Cache/Frontend/Decorator/TagScope.php) class for more info.

Author: Hello @rogyar 
All 15 classes of Magento CE have this message in the PHPDoc. That's why I was pasting it here.

Reviewer: But we always may make things better, aren't we? :) 

From my point of view, the current explanation is not very informative for new developers. Cache type identifier is more or less self-explanatory it's a sort of UID. But the cache tag brings an additional "why?". 

If we have the cache type identifier that distinguishes the cache type from other types why do we need the cache tag that does the same job? 

That's why I think it might be a good idea to clarify this point. I like the explanation [here](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/cache/partial-caching/create-cache-type.html) better. Also, if we could add a word/two (literally) about cache cleaning and cache tag connection it would be very useful for new devs. 

What do you think?

Author: Hello @rogyar 
Of course.
I fixed it.
Please check.
Thank you in advance.

You can get more information about caching by looking at the code.
We suggest you locate classes that extend [Magento\Framework\Cache\Frontend\Decorator\TagScope][tagscope].

> For example, look at [Magento\Customer\Model\Cache\Type\Notification][type] to understand more about the customer notification cache type.

Reviewer: Please, provide in the PR description more info on why exactly the Notification cache type is better for the example than the EAV cache type.

Thank you

Author: Hello @rogyar 
I think it would be more right to use class_name structure like "**_VendorName\ModuleName\Model\Cache\Type\CacheType_**" and most of the classes of Magento CE use it. 12 VS 3. The class "**_Magento\Eav\Model\Cache\Type_**" is contained in these 3 classes. And "**_Magento\Customer\Model\Cache\Type\Notification_**" looks like well on my mind.

I changed the description of this PR.
